backend/handler/database/roms_handler.py

```diff
@@ -606,45 +651,63 @@ def get_roms_scalar(
-            selected_company=kwargs.pop("selected_company", None),
-            selected_age_rating=kwargs.pop("selected_age_rating", None),
-            selected_status=kwargs.pop("selected_status", None),
-            selected_region=kwargs.pop("selected_region", None),
-            selected_language=kwargs.pop("selected_language", None),
-            user_id=kwargs.pop("user_id", None),
+            platform_id=kwargs.get("platform_id", None),
+            collection_id=kwargs.get("collection_id", None),
+            virtual_collection_id=kwargs.get("virtual_collection_id", None),
+            search_term=kwargs.get("search_term", None),
+            matched=kwargs.get("matched", None),
+            favourite=kwargs.get("favourite", None),
+            duplicate=kwargs.get("duplicate", None),
+            playable=kwargs.get("playable", None),
+            has_ra=kwargs.get("has_ra", None),
+            missing=kwargs.get("missing", None),
+            verified=kwargs.get("verified", None),
+            selected_genre=kwargs.get("selected_genre", None),
+            selected_franchise=kwargs.get("selected_franchise", None),
+            selected_collection=kwargs.get("selected_collection", None),
+            selected_company=kwargs.get("selected_company", None),
+            selected_age_rating=kwargs.get("selected_age_rating", None),
+            selected_status=kwargs.get("selected_status", None),
+            selected_region=kwargs.get("selected_region", None),
+            selected_language=kwargs.get("selected_language", None),
+            user_id=kwargs.get("user_id", None),
         )
         return session.scalars(roms).all()
 
     @begin_session
-    def get_char_index(
-        self, query: Query, session: Session = None
+    def with_char_index(
+        self, query: Query, order_by_attr: Any, session: Session = None
     ) -> list[Row[tuple[str, int]]]:
+        if isinstance(order_by_attr.type, (String, Text)):
+            # Remove any leading articles
+            order_by_attr = func.trim(
+                func.lower(order_by_attr).regexp_replace(STRIP_ARTICLES_REGEX, "", "i")
+            )
+        else:
+            order_by_attr = func.trim(
+                func.lower(Rom.name).regexp_replace(STRIP_ARTICLES_REGEX, "", "i")
+            )
+
         # Get the row number and first letter for each item
-        subquery = query.add_columns(
-            func.lower(func.substring(Rom.name, 1, 1)).label("letter"),
-            func.row_number().over(order_by=Rom.name).label("position"),
-        ).subquery()
+        subquery = (
+            query.with_only_columns(Rom.id, Rom.name)  # type: ignore
+            .add_columns(  # type: ignore
+                func.substring(
+                    order_by_attr,
+                    1,
+                    1,
+                ).label("letter"),
+                func.row_number().over(order_by=order_by_attr).label("position"),
+            )
+            .subquery()
+        )
```

Comment:
@zurdi15 to match the order of games we also need to strip leading artiches in the char index query